semgrep/semgrep/rule.py

```diff
@@ -172,11 +179,11 @@ def _parse_boolean_expression(
                         )
 
     @staticmethod
-    def _validate_operand(operand: YamlTree) -> str:  # type: ignore
-        if not isinstance(operand.value, str):
+    def _validate_operand(operand: YamlTree) -> Union[str, dict]:  # type: ignore
+        if not (isinstance(operand.value, str) or isinstance(operand.value, dict)):
             raise InvalidRuleSchemaError(
                 short_msg="invalid operand",
-                long_msg=f"type of `pattern` must be a string, but it was a {type(operand.unroll()).__name__}",
+                long_msg=f"type of `pattern` must be a string or dict, but it was a {type(operand.unroll()).__name__}",
```

Comment:
This may be confusing, as the `dict` presumably won't be in a `pattern` key...

Reply:
I think this should be "type of `operand` ...". The function is `_validate_operand` with `operand` as an argument. I can bump the message